smithers/io/vtuhandler.py

```diff
@@ -50,18 +48,30 @@ def read(cls, filename):
         return result
 
     @classmethod
-    def write(cls, filename, data):
+    def write(cls, vtuOrg, filename, data):
         """
         Method to save the dataset to `filename`. The dataset `data` should be
         a dictionary containing the requested information. The obtained
         `filename` is a well-formatted VTU file.
 
+        :param vtuOrg : original vtu to specify the cell types
         :param str filename: the name of the file to write.
         :param dict data: the dataset to save.
 
-        .. warning:: all the cells will be stored as VTK_POLYGON.
-            
+        .. note::
+          The cells will be saved with the same format as they were,
+          no need to specify the cell type.
         """
+        #VTK object
+        reader = cls._reader_()
+        reader.SetFileName(vtuOrg)
+        reader.Update()
+        vtkObj = reader.GetOutput()
+
+        #define paramaters needed for setCell
+        CellTypes=vtkObj.GetCellTypesArray()
+        faceLoc=vtkObj.GetFaceLocations()
+        face=vtkObj.GetFaces()
```

Comment:
In this way, you are enforced to provide an input VTU even for saving a new one, which is semantically quite chaotic. Why not reading the cellTypes in the `read` method, storing in some attributes, eg `vtk_celltype`, in order to avoid the additional parsing?